include/tvm/ir_operator.h

```diff
@@ -64,38 +462,152 @@ TVM_DECLARE_INTRIN_UNARY(floor);
+  inline Expr Name(Expr& a, Expr b) {                          \
+    a = OpFunc(a, b);                                          \
+    return a;                                                  \
+  }
+
+#define TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(Name)              \
+  inline Expr Name(const Expr& a, float b) {                   \
+    return Name(a, Expr(b));                                   \
+  }                                                            \
+  inline Expr Name(float a, const Expr& b) {                   \
+    return Name(Expr(a), b);                                   \
+  }                                                            \
+  inline Expr Name(int a, const Expr& b) {                     \
+    return Name(make_const(b.type(), a), b);                   \
+  }                                                            \
+  inline Expr Name(const Expr& a, int b) {                     \
+    return Name(a, make_const(a.type(), b));                   \
+  }
+
+#define TVM_DEFINE_LOGICAL_OP_CONST_VAL_OVERLOAD(Name)                  \
+  inline Expr Name(const Expr& a, bool b) {                             \
+    return Name(a, Expr(b));                                            \
+  }                                                                     \
+  inline Expr Name(bool a, const Expr& b) {                             \
+    return Name(Expr(a), b);                                            \
+  }
+
+#define TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(Name)                      \
+  inline Expr Name(const Expr& a, int b) {                              \
+    return Name(a, make_const(a.type(), b));                            \
+  }                                                                     \
+  inline Expr Name(int a, const Expr& b) {                              \
+    return Name(make_const(b.type(), a), b);                            \
+  }
+
+
+TVM_DEFINE_ASSIGN_OP_OVERLOAD(operator+=, operator+);
+TVM_DEFINE_ASSIGN_OP_OVERLOAD(operator-=, operator-);
+TVM_DEFINE_ASSIGN_OP_OVERLOAD(operator*=, operator*);
+TVM_DEFINE_ASSIGN_OP_OVERLOAD(operator/=, operator/);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator+);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator-);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator*);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator/);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(max);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(min);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator>);  // NOLINT(*)
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator>=);
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator<);  // NOLINT(*)
+TVM_DEFINE_BINOP_CONST_VAL_OVERLOAD(operator<=);
+// integer related ops
+TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(operator%);
+TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(operator>>); // NOLINT(*)
+TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(operator<<); // NOLINT(*)
+TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(operator&);
+TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(operator|);
+TVM_DEFINE_INT_OP_CONST_VAL_OVERLOAD(operator^);
```

Comment:
Just curious, do we need `~` op?

Reply:
this macro defines operator for binary ops, ~ is already defined as single operand op